### ISSUE TITLE
DLang Favicon for Twitter etc.

### ISSUE THREAD

**Issue by wilzbach** [dlang/D-Blog-Theme#19]
![image](https://cloud.githubusercontent.com/assets/4370550/16457517/175725c0-3e1c-11e6-9b1c-4083f61bbd4c.png)

Afaik just adding a `favicon.png` would be enough, see this article infos and for common question:

https://github.com/audreyr/favicon-cheat-sheet

there are many such generators out there, e.g. [this](https://gist.github.com/svizion/5989224).

@CyberShadow I guess we could also add this to dlang.org?

**Comment by CyberShadow** [comment]
> @CyberShadow I guess we could also add this to dlang.org?

http://dlang.org/favicon.ico already exists.

**Comment by wilzbach** [comment]
> http://dlang.org/favicon.ico already exists.

Afaik websites like Facebook, Twitter, Google+, ... require a `favicon.png` in the HTML to display an logo on links?

**Comment by CyberShadow** [comment]
dlang.org also has `<link rel="shortcut icon" href="favicon.ico">`.

I think we're doing all the right things. Please make sure that they're not picking it up because of something we need to do.

**Comment by mdparker** [comment]
This has nothing to do with the favicon or anything outside of the specific blog posts. See the section 'Images in Publicize'.

https://jetpack.com/support/publicize/

I suppose what I can do is see about getting a default image for each post if there are none in the post itself. I'll look into it.

**Comment by wilzbach** [comment]
> This has nothing to do with the favicon or anything outside of the specific blog posts.

Oh sorry - yeah it seems like all these website have special `meta` formats.

https://dev.twitter.com/cards/types/summary

``` html
<meta name="twitter:card" content="summary" />
<meta name="twitter:site" content="@flickr" />
<meta name="twitter:title" content="Small Island Developing States Photo Submission" />
<meta name="twitter:description" content="View the album on Flickr." />
<meta name="twitter:image" content="https://farm6.staticflickr.com/5510/14338202952_93595258ff_z.jpg" />
```

``` html
<meta property="og:url"                content="http://www.nytimes.com/2015/02/19/arts/international/when-great-minds-dont-think-alike.html" />
<meta property="og:type"               content="article" />
<meta property="og:title"              content="When Great Minds Don’t Think Alike" />
<meta property="og:description"        content="How much does culture influence creative thinking?" />
<meta property="og:image"              content="http://static01.nyt.com/images/2015/02/19/arts/international/19iht-btnumbers19A/19iht-btnumbers19A-facebookJumbo-v2.jpg" />
```

https://developers.facebook.com/docs/sharing/webmasters

@CyberShadow a `favicon.png` is useful anyways as it has a higher resolution and most browser can use it (e.g. bookmarks, tiles, Android home shortcuts ...). See the cheatsheet for a list of resolutions required by specific devices:

https://github.com/audreyr/favicon-cheat-sheet

**Comment by CyberShadow** [comment]
To be honest, that page makes me want to barf. Ten thousand formats for each individual browser. Fuck that.

**Comment by mdparker** [comment]
I finally looked into this. I've just configured WP with a default featured image (a.k.a thumbnail). From now on, if a post has no embedded images, or I forget to insert one before I publish it (as I did with the Visual D post, so that it went to Twitter/FB without an image, which is what I wanted to avoid int he first place!), then the default will be used when it's sent out to Twitter/FB. And, I suppose, anywhere else people may share it.
